dataprofiler/data_readers/json_data.py

```diff
@@ -51,83 +56,201 @@ def __init__(self, input_file_path=None, data=None, options=None):
+        """
+        data = self.data
+        if self._metadata is not None and not self._metadata.empty:
+            data = [self._metadata, data]
+            data = pd.concat(data, axis=1)
+        return data
+
+    def _find_data(self, json_data, path=""):
+        """
+        Finds all the column headers/data in a Json and returns them as a 
+        list.
+
+        :param json_data: the json data or a subset of the json data
+        :type json_data: Union[list, dict, str]
+        :param path: path of keys to the json_data
+        :type path: str
+        :return: list of dicts of {column headers: values}
+        """
+        if path != "" and path[-len(self._key_separator):] != self._key_separator:
+            path = path + self._key_separator
+
+        list_of_dict = []
+        if isinstance(json_data, dict):
+            for key in json_data:
+                if isinstance(json_data[key], dict) or isinstance(json_data[key], list):
+                    list_of_dict = list_of_dict + self._find_data(json_data[key], path + key)
+                else:
+                    list_of_dict = list_of_dict + [{path + key: json_data[key]}]
+        elif isinstance(json_data, list):
+            if all(isinstance(x, dict) for x in json_data):
+                for key in json_data:
+                    list_of_dict = list_of_dict + self._find_data(key, path)
+            else:
+                list_of_dict = list_of_dict + [{path[:-len(self._key_separator)]: json_data}]
+        else:
+            list_of_dict = list_of_dict + [{path[:-len(self._key_separator)]: json_data}]
+        return list_of_dict
+    
+    def _coalesce_dicts(self, list_of_dicts):
+        """
+        Merges all the dictionaries into as few dictionaries as possible.
+
+        :param list_of_dicts: the list of dictionaries to be coalesced
+        :type list_of_dicts: list(dict)
+        :return: Coalesced list of dictionaries
+        """
+        coalesced_list_of_dicts = [{}]
+        for item in list_of_dicts:
+            found = False
+            for dict_items in coalesced_list_of_dicts:
+                if list(item.keys())[0] not in dict_items:
+                    dict_items.update(item)
+                    found = True
+                    break
+            if not found:
+                coalesced_list_of_dicts.append(item)
+        return coalesced_list_of_dicts
+
+    def _get_data_as_flattened_dataframe(self, json_lines):
+        """
+        Loads the data when in a JSON format from a data stream (nested
+        lists of dictionaries and a key value for a payload.
+
         :param json_lines: json format list of dicts or dict
         :return:
         """
+        if isinstance(json_lines, pd.DataFrame):
+            return json_lines
+        payload_data = None
         if isinstance(json_lines, dict):
-            json_lines = [json_lines]
+            # Glean Payload Data
+            found_payload_key = None
```

Comment:
What if multiple payloads found?

Should probably add a test and clarify what'll happen here.

Reply:
Just takes the first payload it finds